alacritty_terminal/src/tty/windows/child.rs

```diff
@@ -85,6 +90,21 @@ impl ChildExitWatcher {
     pub fn deregister(&self) {
         *self.interest.lock().unwrap() = None;
     }
+
+    /// Retrieve the process handle of the underlying child process.
+    /// This function does **not** pass ownership of the raw handle to you,
+    /// and the handle is only guaranteed to be valid while the hosted application
+    /// has not yet been destroyed.
+    /// If you terminate the process using this handle, the terminal will get a
+    /// timeout error, and the child watcher will emit an `Exited` event.
+    pub fn raw_handle(&self) -> HANDLE {
+        self.child_handle
+    }
+
+    /// Retrieve the Process ID associated to the underlying child process.
+    pub fn pid(&self) -> u32 {
```

Comment:
Maybe we should use `Option<NonZeroU32>` for that? Since the value is 0 when it errors and it generally can't be _zero_. Though, I'm not strong on that.

Reply:
> Maybe we should use `Option<NonZeroU32>` for that? Since the value is 0 when it errors and it generally can't be _zero_. Though, I'm not strong on that.

Oh, I forgot the error handling here, sorry.